### PR TITLE
Remove FAQ page and add info page about demo instance

### PR DESCRIPTION
**Pull request by ErikMichelson**
### Description
This PR removes the FAQ page as that will be moved to the docs.
To answer the questions about the demo instance, there will be an extra page about the demo instance. This is intended to work as an interstitial page showing users why they should not rely on the demo instance.

### Steps
- [x] Added changes
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc.github.io/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### Related Issue(s)
hedgedoc/hedgedoc#1532

### Additional information
Deployed at https://webpage-staging-149.hedgedoc.net/

### CODE REVIEW

**Comment by mrdrogdrog** [comment]
Why should we move the FAQ to the docs?

**Comment by ErikMichelson** [comment]
We discussed that the contents of the FAQ are more docs-relevant. And we wanted to avoid having things in two places.

**Comment by mrdrogdrog** [comment]
When and where

**Comment by davidmehren** [comment]
@mrdrogdrog https://github.com/hedgedoc/hedgedoc/pull/1532 adds more technical questions to the FAQ.
Also, the FAQs on the homepage can't be searched by the docs search. Therefore, moving the FAQ to the docs improves the user experience.

File: content/english/demo.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+draft: false
+---
+
+The demo instance is a publicly available instance for trying out HedgeDoc and can be found at [demo.hedgedoc.org](https://demo.hedgedoc.org).
+
+While notes and accounts on the demo instance don't get deleted periodically, **you should not rely on the demo instance for production use**.  
+**We cannot guarantee that your notes won't be lost** because of hardware or software malfunctions.
+This is mainly due to the fact that this instance runs the latest unstable code from our repository.
+
+Some countries have a too high latency to the demo instance, which means HedgeDoc would be unusable. To make the demo available in as many countries as possible, we use [Cloudflare](https://www.cloudflare.com/). One of the [downsides](https://en.wikipedia.org/wiki/Cloudflare#Controversy) is that Cloudflare intercepts all TLS traffic between your browser and the demo instance, which makes the demo unsuitable for private notes.
```
Comment (DerMolly) [inline content/english/demo.md:13]: I think this sentence needs some rearrangement. Maybe start with `To make the demo available…`

**Review by DerMolly — CHANGES_REQUESTED** [review]

**Comment by mrdrogdrog** [comment]
Don't forget to add the faq page AND the draft faq entries to the develop branch!

**Review by mrdrogdrog — APPROVED** [review]

**Review by DerMolly — APPROVED** [review]

**Review by davidmehren — APPROVED** [review]

**Pull request merged**